### PR TITLE
feat(wallet): resize saved address popup to remove network selection

### LINKED ISSUE #13381: Ability to add ENS domain name as a saved address
Creating saved address flows updated to include the case of adding an ENS domain as a saved address.

The user should be able to enter an ENS domain into the `Address` field. The resolver service should ensure the appropriate address is used for the ENS at the point of interaction (e.g. sending tokens to the ENS domain).

Screenshot 2024-02-01 at 2 50 37 PM

**Figma**
https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?type=design&node-id=23256-263282&mode=design&t=0DRwQJKDGYJPHkq1-4

### PR DESCRIPTION
**Pull request by IvanBelyakoff**
when ENS name is resolved.
- Resolve ENS name before used in SendModal
- UI tweaks:
  - red stroke on address input in case of error
  - smaller tick for validation address input
  - added validation spinner to address input, removed from the button (Ben updated design)
  - handled tab key to move focus between inputs
  - animation for resizing the dialog

### Affected areas

Saved address popup

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [x] update storybook app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/15627093/39604d4a-a6af-4f10-a6b2-c8f1c492148c

[x ] I've checked the design and this PR matches it

Closed #13381

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml [hunk 1]
```diff
@@ -273,7 +276,16 @@ StatusModal {
...
             d.address = resolvedAddress
-            d.checkForAddressInputOwningErrorsWarnings()
+            try { // allows to avoid issues in storybook without much refactoring
+                d.checkForAddressInputOwningErrorsWarnings()
+            }
+            catch (e) {
+            }
+
+            if (!d.addressInputHasError)
+                networkSelector.state = "networksHidden"
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:286]: @IvanBelyakoff could you set this as readonly property?

`readonly property string networksHiddenState: "networksHidden"`

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml [hunk 2]
```diff
@@ -612,6 +628,27 @@ StatusModal {
...
+                    State {
+                        name: "networksHidden"
+                        PropertyChanges { target: networkSelector; height: 0 }
+                        PropertyChanges { target: networkSelector;  opacity: 0 }
+                        PropertyChanges { target: column; bottomPadding: 0 }
+                    }
+                ]
+                transitions: [
+                    Transition {
+                        PropertyAnimation { property: "height"; duration: 250 }
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:644]: Could you set duration as readonly property?

**Review by saledjenic — APPROVED** [review]
@IvanBelyakoff changes look good, just 2 things:
- we're still unsure whether to add chain prefixes for ens names or not (cause in the end that's just an alias for a real address)
- if we decide to don't add them then we need to update show QR popup similarly to what you did here

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml [hunk 3]
```diff
@@ -612,6 +629,27 @@ StatusModal {
...
+                        PropertyChanges { target: column; bottomPadding: 0 }
+                    }
+                ]
+                transitions: [
+                    Transition {
+                        PropertyAnimation { property: "height" }
+                        PropertyAnimation { property: "opacity" }
+                        SequentialAnimation {
+                            PropertyAnimation { property: "bottomPadding" }
+                            PropertyAction { target: column; property: "bottomPadding"; value: 28 }
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:649]: Pls put the `28` magic number into some constant :)

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml [hunk 4]
```diff
@@ -273,7 +277,16 @@ StatusModal {
...
             d.resolvingEnsNameInProgress = false
             d.address = resolvedAddress
-            d.checkForAddressInputOwningErrorsWarnings()
+            try { // allows to avoid issues in storybook without much refactoring
+                d.checkForAddressInputOwningErrorsWarnings()
+            }
+            catch (e) {
+            }
+
+            if (!d.addressInputHasError)
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:286]: You could remove this imperative piece of JS code and replace it with...

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml [hunk 5]
```diff
@@ -612,6 +629,27 @@ StatusModal {
                     item.modelRef.isEnabled = !item.modelRef.isEnabled
                     d.chainShortNamesDirty = true
                 }
+
+                states: [
+                    // As when networks seclector becomes invisible, spacing before it disappears as well, we see jumping height
+                    // To overcome this, we animate bottom padding to 0 and when spacing disappears, reset bottom padding back to compensate it
+                    State {
+                        name: d.networksHiddenState
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:637]:
```suggestion
                        name: d.networksHiddenState
                        when: !d.addressInputHasError
```

Comment (caybro) [reply ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:637]: ... replace it with `when`

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml (hunk 4 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:286]: Fair enough!

**Review by caybro — APPROVED** [review]
LGTM :+1:

**Pull request merged**